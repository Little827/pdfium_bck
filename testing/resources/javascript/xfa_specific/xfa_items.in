{{header}}
{{include ../../xfa_catalog_1_0.fragment}}
{{include ../../xfa_object_2_0.fragment}}
{{include ../../xfa_preamble_3_0.fragment}}
{{include ../../xfa_config_4_0.fragment}}
{{object 5 0}} <<
  {{streamlen}}
>>
stream
<template xmlns="http://www.xfa.org/schema/xfa-template/2.6/">
  <subform name="form1">
    <pageSet>
      <pageArea id="Page1" name="Page1">
        <contentArea h="10.5in" w="8in" x="0.25in" y="0.25in"/>
        <medium long="11in" short="8.5in" stock="letter"/>
      </pageArea>
    </pageSet>
    <field h="9.0001mm" name="field1" w="47.625mm" x="6.35mm" y="92.075mm">
      <ui>
        <choiceList/>
      </ui>
      <items nonesuch="3">
        <arc name="arc1"></arc>
        <boolean name="bool0">0</boolean>
        <boolean name="bool1">1</boolean>
        <boolean name="boolbad">bad</boolean>
        <boolean name="booltruenottrue">true</boolean>
        <date name="date0"></date>
        <date name="date1">2020-02-02</date>
        <date name="date2">2039-12-01</date>
        <date name="datebad">bad</date>
        <dateTime name="datetime0"></dateTime>
        <dateTime name="datetime1">2020-02-02T12:34:56</dateTime>
        <dateTime name="datetime2">2039-12-01T12:34:56</dateTime>
        <dateTime name="datetimebad">bad</dateTime>
        <decimal name="decimal0"></decimal>
        <decimal name="decimal1">42.0000000000000000001</decimal>
        <decimal name="decimalbad">bad</decimal>
        <exData name="ex0"></exData>
        <exData name="ex1"><![CDATA[YZYZYZYZYZYZYZYZYZYZYZYZYZYZ]]></exData>
        <float name="float0">-12.34</float>
        <float name="float1">-12.34</float>
        <float name="floatbad">bad</float>
        <image name="image0">ABABABABABABABABA</image>
        <image name="image1"><![CDATA[ABABABABABABABABA]]></image>
        <integer name="int0"></integer>
        <integer name="int1">1234</integer>
        <integer name="intbad">bad</integer>
        <line name="line0"></line>
        <rectangle name="rect0"></rectangle>
        <text name="text0"></text>
        <text name="text1">Ahoy !!!</text>
        <time name="time0"></time>
        <time name="time1">12:34:56</time>
      </items>
    </field>
    <event activity="docReady">
      <script contentType="application/x-javascript"><![CDATA[
        {{include ../expect.js}}
        {{include dump_tree.js}}

        itemlist = xfa.resolveNode("form1.field1.#items");
        dumpTree(itemlist);

        arc1 = itemlist.resolveNode("arc1");

        bool0 = itemlist.resolveNode("bool0");
        expect("bool0.value", false);
        bool0.value = 1;
        expect("bool0.value", true);

        bool1 = itemlist.resolveNode("bool1");
        expect("bool1.value", true);
        bool1.value = 0;
        expect("bool1.value", false);

        boolbad = itemlist.resolveNode("boolbad");
        expect("boolbad.value", false);

        booltruenottrue = itemlist.resolveNode("booltruenottrue");
        expect("booltruenottrue.value", false);

        // TODO(tsepez): confirm if this is correct.
        booltruenottrue.value = true;
        expect("booltruenottrue.value", false);

        booltruenottrue.value = "zerp";
        expect("booltruenottrue.value", false);
        booltruenottrue.value = "1";
        expect("booltruenottrue.value", true);
        booltruenottrue.value = "10";
        expect("booltruenottrue.value", true);
        booltruenottrue.value = "1zerp";
        expect("booltruenottrue.value", true);

        // Date is just a node, and allows any text within.
        date0 = itemlist.resolveNode("date0");
        expect("date0.value", null);

        date1 = itemlist.resolveNode("date1");
        expect("date1.value", "2020-02-02");

        date2 = itemlist.resolveNode("date2");
        expect("date2.value", "2039-12-01");

        datebad = itemlist.resolveNode("datebad");
        expect("datebad.value", "bad");

        // These are pretty much just nodes, and allow any text within.
        datetime0 = itemlist.resolveNode("datetime0");
        datetime1 = itemlist.resolveNode("datetime1");
        datetime2 = itemlist.resolveNode("datetime2");
        datetimebad = itemlist.resolveNode("datetimebad");
        decimal0 = itemlist.resolveNode("decimal0");
        decimal1 = itemlist.resolveNode("decimal1");
        decimalbad = itemlist.resolveNode("decimalbad");
        ex0 = itemlist.resolveNode("ex0");
        ex1 = itemlist.resolveNode("ex1");
        float0 = itemlist.resolveNode("float0");
        float1 = itemlist.resolveNode("float1");
        floatbad = itemlist.resolveNode("floatbad");
        image0 = itemlist.resolveNode("image0");
        image1 = itemlist.resolveNode("image1");
        int0 = itemlist.resolveNode("int0");
        int1 = itemlist.resolveNode("int1");
        intbad = itemlist.resolveNode("intbad");
        line0 = itemlist.resolveNode("line0");
        rect0 = itemlist.resolveNode("rect0");
        text0 = itemlist.resolveNode("text0");
        text1 = itemlist.resolveNode("text1");
        time0 = itemlist.resolveNode("time0");
        time1 = itemlist.resolveNode("time1");
      ]]></script>
    </event>
  </subform>
</template>
endstream
endobj
{{include ../../xfa_locale_6_0.fragment}}
{{include ../../xfa_postamble_7_0.fragment}}
{{include ../../xfa_pages_8_0.fragment}}
{{xref}}
{{trailer}}
{{startxref}}
%%EOF
