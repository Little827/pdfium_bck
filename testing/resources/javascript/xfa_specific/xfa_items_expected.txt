Alert: items
Alert: | arc
Alert: | boolean
Alert: | boolean
Alert: | boolean
Alert: | boolean
Alert: | date
Alert: | date
Alert: | | #text
Alert: | date
Alert: | | #text
Alert: | date
Alert: | | #text
Alert: | dateTime
Alert: | dateTime
Alert: | | #text
Alert: | dateTime
Alert: | | #text
Alert: | dateTime
Alert: | | #text
Alert: | decimal
Alert: | decimal
Alert: | decimal
Alert: | exData
Alert: | exData
Alert: | float
Alert: | float
Alert: | float
Alert: | image
Alert: | | #text
Alert: | image
Alert: | | #text
Alert: | integer
Alert: | integer
Alert: | integer
Alert: | line
Alert: | rectangle
Alert: | text
Alert: | text
Alert: | time
Alert: | time
Alert: PASS: bool0.value = false
Alert: PASS: bool0.value = true
Alert: PASS: bool1.value = true
Alert: PASS: bool1.value = false
Alert: PASS: boolbad.value = false
Alert: PASS: booltruenottrue.value = false
Alert: PASS: booltruenottrue.value = false
Alert: PASS: booltruenottrue.value = false
Alert: PASS: booltruenottrue.value = true
Alert: PASS: booltruenottrue.value = true
Alert: PASS: booltruenottrue.value = true
Alert: PASS: date0.value = null
Alert: PASS: date1.value = 2020-02-02
Alert: PASS: date2.value = 2039-12-01
Alert: PASS: datebad.value = bad
Alert: FAIL: datetime0 = [object Object], expected [object XFAObject] 
Alert: FAIL: datetime1 = [object Object], expected [object XFAObject] 
Alert: FAIL: datetime2 = [object Object], expected [object XFAObject] 
Alert: FAIL: datetimebad = [object Object], expected [object XFAObject] 
Alert: FAIL: decimal0 = [object Object], expected [object XFAObject] 
Alert: FAIL: decimal1 = [object Object], expected [object XFAObject] 
Alert: FAIL: decimalbad = [object Object], expected [object XFAObject] 
Alert: FAIL: ex0 = [object Object], expected [object XFAObject] 
Alert: FAIL: ex1 = [object Object], expected [object XFAObject] 
Alert: FAIL: float0 = [object Object], expected [object XFAObject] 
Alert: FAIL: float1 = [object Object], expected [object XFAObject] 
Alert: FAIL: floatbad = [object Object], expected [object XFAObject] 
Alert: FAIL: image0 = [object Object], expected [object XFAObject] 
Alert: FAIL: image1 = [object Object], expected [object XFAObject] 
Alert: FAIL: int0 = [object Object], expected [object XFAObject] 
Alert: FAIL: int1 = [object Object], expected [object XFAObject] 
Alert: FAIL: intbad = [object Object], expected [object XFAObject] 
Alert: FAIL: line0 = [object Object], expected [object XFAObject] 
Alert: FAIL: rect0 = [object Object], expected [object XFAObject] 
Alert: FAIL: text0 = [object Object], expected [object XFAObject] 
Alert: FAIL: text1 = [object Object], expected [object XFAObject] 
Alert: FAIL: time0 = [object Object], expected [object XFAObject] 
Alert: FAIL: time1 = [object Object], expected [object XFAObject] 
Alert: PASS: goop0 = null
